src/pure-assert.js

```diff
@@ -55,40 +55,118 @@ export class UserError extends Error {
  * Throws a provided error if the second argument isn't trueish.
  * @param {Object} errorCls
  * @param {T} shouldBeTruthy
- * @param {string} message
+ * @param {string} opt_message
+ * @param {...*} var_args Arguments substituted into %s in the message
  * @return {T}
- * @throws {Error} when attribute values are missing or invalid.
+ * @throws {Error} when shouldBeTruthy is not truthy.
  */
-function pureAssertion(errorCls, shouldBeTruthy, message) {
+function pureAssertion(errorCls, shouldBeTruthy, opt_message, var_args) {
   // TODO: Support format strings.
-  if (!shouldBeTruthy) {
-    throw new errorCls(message);
+  if (shouldBeTruthy) {
+    return shouldBeTruthy;
   }
-  return shouldBeTruthy;
+
+  // Substitute provided values into format string in message
```

Comment:
Is the `TODO: Support format strings.` above not solved by this code?

Reply:
Removed, thank you